108 integrate frontend with backend apis

oduortoni opened the pull request:
## Overview

This PR enables the /me page to fetch data from the backend. It adds real-time user presence tracking and enhances the notification system. The `/me` page now shows currently online users with nicknames and avatars, updated live via WebSocket.

It has included all the changes from the:

1. development

2. 112-implement-follow-and-unfollow-backend-logic

branches, in order to avoid conflict

## Features

### Real-Time Presence

* Displays live list of connected users on `/me`.
* Includes full profile info: `nickname` and `avatar`.
* Connection and disconnection events are broadcast system-wide.

### Improved Notifications

* Notifications are sent when users come online or go offline.
* Bell icon shows unread connection notifications.
* Notification drawer tracks read/unread state.
* WebSocket client now supports multiple handlers per message type.

### Avatar & Nickname Support

* Avatar and nickname now persist from login through all WebSocket activity.
* Connection messages and user lists now include avatar URLs.
* Users with no avatar fallback to a clean base64 SVG circle avatar.

## Technical Improvements

* Sessions now resolve to `user_id`, `nickname`, and `avatar`.
* WebSocket ping/pong every 30 seconds to detect stale connections.
* Consistent data structures used across client and server.
* Fixed race conditions and handler collisions in real-time updates.

## Example Payload

```json
{
  "type": "notification",
  "subtype": "user_connected",
  "user_id": 123,
  "nickname": "Jane",
  "avatar": "avatar.jpg",
  "message": "Jane is now online",
  "timestamp": 1721990400
}
```

oduortoni committed:
feat: Integrate user profile data

- Pass user data from /me page through MainHomepage to Header component
- Display actual user avatar and nickname in Header instead of hardcoded values
- Update ProfileSidebar to use real user data from props
- Add profileAPI integration for dynamic avatar image fetching
- Update ProfileSidebar to display user's nickname and avatar

- Dynamic profile display showing authenticated user's actual information
- Real-time user data integration across header and sidebar components
- Proper user avatar handling with fallback support

- Improved component prop passing for user data consistency

This commit establishes the foundation for user profile display and social network
relationship management features while maintaining clean component architecture.

Breaking Changes: None
Migration Required: None (additive changes only)

oduortoni committed:
feat: Add avatar support to WebSocket system and enhance connected users tracking

- **Enhanced session resolver** to return user avatar alongside ID and nickname
- **Updated WebSocket client structure** to include avatar field for richer user data
- **Modified notification system** to broadcast user avatars in connection events
- **Fixed all WebSocket tests** to support new avatar-enabled schema

- **Connected users tracking** with real-time updates via WebSocket notifications
- **Avatar support** in user data flow from authentication through components
- **Enhanced Feed component** to receive user and connected users data
- **Improved MainHomepage** to pass connected users to Feed component

- **Online users endpoint** now returns avatar data for connected users
- **Real-time notifications** include user avatars for better UX
- **WebSocket connection events** broadcast complete user profile data

- **Updated test schemas** to include avatar fields in users table
- **Fixed integration tests** with proper avatar test data
- **Enhanced WebSocket documentation** with avatar support examples
- **All tests passing** with comprehensive avatar integration

Connected users now return: `{user_id, nickname, avatar, connected_timestamp}`

This enables rich user presence indicators and avatar display throughout the social network interface.

Breaking Changes: None (additive avatar support)
Migration Required: Database schema already supports avatars

oduortoni committed:
docs: Add comprehensive project requirements and real-time strategy documentation

- **REQUIREMENTS.md**: Complete project specification tracking with progress status for all features (authentication, profiles, posts, groups, chat, notifications)
- **RealtimeUpdates.md**: Detailed real-time implementation strategy with server-side ping/pong, notification-driven content updates, and hybrid WebSocket/polling approach

- Real-time for chat, notifications, presence; notification-driven for content updates
- Server-initiated ping for reliable connection health monitoring
- Smart client-side refresh based on user context and activity

Establishes clear technical roadmap and implementation guidelines for social network development.

oduortoni committed:
fix: Resolve real-time notifications and field name consistency issues

- **Fixed multiple message handlers**: WebSocket service now supports multiple handlers per message type instead of overwriting
- **Restored connection notifications**: User connect/disconnect notifications now work properly in both header bell and connected users list

- **Unified on 'nickname'**: Fixed frontend components using inconsistent 'user_name' vs 'nickname' field names
- **Updated documentation**: All examples now consistently use 'nickname' field to match backend implementation

- ChatInterface, /me page, notification handlers now use consistent 'nickname' field
- Real-time user presence tracking fully functional with proper data flow

Resolves notification system conflicts and ensures consistent user data handling across frontend/backend.

oduortoni committed:
Merge remote-tracking branch 'origin/112-implement-follow-and-unfollow-backend-logic' into temp

oduortoni committed:
Merge pull request #215 from tajjjjr/temp

Temp

oduortoni committed:
fix: remove debug logs

oduortoni committed:
feat: design the activity sidebar message format

Introduce a way to have a sidebar that can house various types of messages
By creating context-aware messages, we can provide users with relevant information without overwhelming them with too much data -- by virtue of extra screen designs
This is a good way to provide users with a clear and concise overview of their current state and what not
Essentially simplicity by complex abstraction

oduortoni committed:
Merge branch 'development' into 108-integrate-frontend-with-backend-apis

oduortoni committed:
docs: recent activities

Documentation for the planned "Recent Activities" section
We are trying to make mesages generic so that they can all be housed within one div

oduortoni committed:
feat: refactor websocket folder

remove the handlers and take them to the handlers folder
remove any http related code from websocket folder

This PR makes the websocket functionality adhere to the file structure

oduortoni committed:
Merge pull request #217 from tajjjjr/recent-activity

Clean up the websocket code

nyunja reviewed (approved):
fix the merge issues

oduortoni committed:
Merge branch 'development' into 108-integrate-frontend-with-backend-apis

oduortoni committed:
fix: merge issues in Feed.jsx

Added PostList and CreatePost components, which were deleted by git's automatic merging process.

Siaka385 reviewed (approved):
merging issue fixed, pr can be merged
